File: clojure/grains/src/grains.clj
(ns grains)

(defn square
  "Returns the number of grains on the n-th chessboard square."
  [n]
  (reduce *' (repeat (dec n) 2)))

(defn total
  "Returns the total number of grains on the chessboard."
  []
  (dec (square 65)))

File: python/grains/grains.py
"""Functions to calculate the number of grains of wheat on a chessboard"""

def square(number):
    if number < 1 or number > 64:
        raise ValueError('square must be between 1 and 64')
    return 2 ** (number - 1)


def total():
    return sum(square(num) for num in range(1, 65))
